### PR TITLE
[Main Course] - Tabnet

### PR DESCRIPTION
**Pull request by jhqian0909**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Post for Tabnet

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Tina0330 — APPROVED** [review]
LGTM，感谢佳辉分享，又是做笔记的一天 :)

File: docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/tabnet3.png" width="500" />
+</figure>
+
+特征先通过一个BN(batch normalization)层，接着通过Feature transformer层。这一层的作用和之前所述的全连接层类似，都是在做特征计算，它的结构如下：
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/tabnet4.png" width="500" />
+</figure>
+
+它有两大部分，所有step共享的部分以及每个step独有的决策部分，其中的小结构都是残差连接的由全连接层，BN层以及用来做特征选择的GLU(Gated Linear Unit)构成的小结构。接着split后(选择部分特征)，通过attentive transforme进行特征选择(主要通过sparsemax将某些特征置为0)。直观上可以理解为之前被多个steps选中的特征就不应该被模型选择了。最终生成运用于下个step的Mask矩阵。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md:53]: 全连接层（FC），这个缩写在图上有，文字中没写

Comment (Realvincentyuan) [reply docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md:53]: attentive transformer，少了一个r

**Review by Realvincentyuan — APPROVED** [review]
LGTM

File: docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    作者：谁是Void，发布于2021-12-18，阅读时间：约6分钟，微信公众号文章链接：[:fontawesome-solid-link:](https://mp.weixin.qq.com/s?__biz=MzI4Mjk3NzgxOQ==&mid=2247484578&idx=1&sn=54593557289d011ac6aff472597a731b&chksm=eb90f7d6dce77ec0d896ff1149f89f4c6d9bab7753be123ef269aa781a3ae88012bfa90ce2e0&token=221998215&lang=zh_CN#rd)
+
+## 1 前言
+
+对于表格型数据，树模型(LightGBM，XGBoost)往往能有不错的表现。可能的原因有:
+
+- 容易构造或已经有丰富的特征库。
+- 树模型的决策流形(decision manifolds)是超平面边界的(可以理解为一刀一刀切出来的)，对此类问题表现较好。
+
+由于NN模型(Neural Network Model)的表现只能算差强人意，而我们又需要有NN模型参与最后的模型ensemble，机智的研究者们为此设计出了类似树模型的NN模型。本文将介绍的就是此类模型：TabNet。
```
[the remark this answers is not in the document]
Comment (jhqian0909) [reply docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md]: 已修改

File: docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md (hunk 1 above)
Comment (jhqian0909) [reply docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md]: 已修改

Comment (EchoTang04) [inline docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md]: “其中的小结构都是残差连接的由全连接层，BN层以及用来做特征选择的GLU(Gated Linear Unit)构成的小结构”，这句话有点没读懂。是不是每个小结构都是FC+BN+GLU组成，然后小结构之间是残差连接的？

**Review by EchoTang04 — APPROVED** [review]
LGTM, 感谢佳辉分享！

File: docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md (hunk 1 above)
Comment (jhqian0909) [reply docs/Main_Course/Deep_Learning/2021-12-18-Tabnet.md]: 是的

**Pull request merged**